hummingbot/connector/exchange/digifinex/digifinex_api_user_stream_data_source.py

```diff
@@ -102,12 +50,31 @@ async def listen_for_user_stream(self, output: asyncio.Queue):
 
         while True:
             try:
-                async for msg in self._listen_to_orders_trades_balances():
+                self._ws = DigifinexWebsocket(self._global.auth)
+                await self._ws.connect()
+
+                trading_pairs: List[str] = [digifinex_utils.convert_to_ws_trading_pair(pair)
+                                            for pair in self._trading_pairs]
+                currencies = list()
+                for trade_pair in self._trading_pairs:
+                    trade_pair_currencies = trade_pair.split('-')
+                    currencies.extend(trade_pair_currencies)
+
+                await self._ws.subscribe("order", trading_pairs)
+                await self._ws.subscribe("balance", currencies)
+
+                async for msg in self._ws.iter_messages():
+                    if msg is None or "params" not in msg or "method" not in msg:
+                        continue
                     output.put_nowait(msg)
             except asyncio.CancelledError:
                 raise
-            except Exception:
+            except ClientConnectionError:
+                self.logger().warning("Attemping re-connection with Websocket Private Channels...")
+            except Exception as e:
                 self.logger().error(
-                    "Unexpected error with Digifinex WebSocket connection. " "Retrying after 30 seconds...", exc_info=True
+                    f"Unexpected error with WebSocket connection. {str(e)}",
+                    exc_info=True
                 )
-                await asyncio.sleep(30.0)
+            finally:
+                await self._ws.disconnect()
```

Comment:
Here we should add the same check that self._ws is not None, as in the order book data source